src/main/java/io/hops/metadata/hdfs/entity/FileProvXAttrBufferEntry.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2019 hops.io.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hops.metadata.hdfs.entity;
+
+import com.google.common.base.Charsets;
+
+public final class FileProvXAttrBufferEntry {
+
+  private final long iNodeId;
+  private final byte namespace;
+  private final String name;
+  private final int iNodeLogicalTime;
+  private final String value;
+
+  public FileProvXAttrBufferEntry(long iNodeId, byte namespace, String name, int iNodeLogicalTime, String value) {
+    this.iNodeId = iNodeId;
+    this.namespace = namespace;
+    this.name = name;
+    this.iNodeLogicalTime = iNodeLogicalTime;
+    this.value = value;
+  }
+
+  public FileProvXAttrBufferEntry(long inodeId, byte namespace, String name, int inodeLogicalTime, byte[] value) {
+    this(inodeId, namespace, name, inodeLogicalTime, getXAttrString(value));
+  }
+
+  public long getInodeId() {
+    return iNodeId;
+  }
+
+  public byte getNamespace() {
+    return namespace;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public int getINodeLogicalTime() {
+    return iNodeLogicalTime;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  public byte[] getValueBytes() {
+    return getXAttrBytes(value);
+  }
+
+  public final static byte[] getXAttrBytes(String val) {
```

Comment:
Same as below.
return StoredXAttr. getXAttrBytes(val)

Reply:
fixed.
I removed the static methods and used your methods where i was using my own before.